wire-ios-request-strategy/Sources/Synchronization/Decoding/StoreUpdateEventTests.swift

```diff
@@ -271,6 +271,41 @@ class StoreUpdateEventTests: MessagingTestBase {
         }
     }
 
+    func test_EncryptAndCreate_DoesNotStoreDuplicateEvents() throws {
+        eventMOC.performAndWait {
+            // Given some events.
+            let conversation = self.createConversation(in: self.uiMOC)
+            let event1 = self.createNewConversationEvent(for: conversation)
+
+            // When we store the events with public keys.
+            guard let storedEvent1 = StoredUpdateEvent.encryptAndCreate(
+                event1,
+                context: self.eventMOC,
+                index: 1,
+                publicKeys: nil
+            ) else {
+                return XCTFail("Did not create storedEvent")
+            }
+
+            let storedEvent2 = StoredUpdateEvent.encryptAndCreate(
+                event1,
+                context: self.eventMOC,
+                index: 2,
+                publicKeys: nil
+            )
```

Comment:
This is probably the same as @johnxnguyen meant.

By creating a second event in a different NSDictionary instance with the same content you will see that the test fails, because NSDictionary doesn't conform to `Hashable` properly, in my opinion.

```suggestion
            let event1 = self.createNewConversationEvent(for: conversation)
            let event2 = self.createNewConversationEvent(for: conversation) // <---

            // When we store the events with public keys.
            guard let storedEvent1 = StoredUpdateEvent.encryptAndCreate(
                event1,
                context: self.eventMOC,
                index: 1,
                publicKeys: nil
            ) else {
                return XCTFail("Did not create storedEvent")
            }

            let storedEvent2 = StoredUpdateEvent.encryptAndCreate(
                event2, // <---
                context: self.eventMOC,
                index: 2,
                publicKeys: nil
            )
```